### ISSUE TITLE
[P2][universal-platform] import core.engine SIGILL (rc132) em CPU sem AVX: baseline do wheel numpy > CPU — falha SILENCIOSA (G1-é-silent), ameaça tese hardware-velho

### ISSUE THREAD

**Issue by FabioLeitao** [DataBoar/data-boar#929]
## Evidência (host alpine — musl x64, eMachines velho, **CPU SEM avx**)
`import core.engine` morre com **SIGILL (rc 132)** — crash NATIVO, **zero traceback Python**. Bisect isola o root:
```
import numpy            -> rc 132 (SIGILL)
import scipy            -> rc 132   (importa numpy)
import sklearn          -> rc 132   (importa numpy)
import core.dl_backend  -> rc 132   (importa numpy)
import core.engine      -> rc 132   (cadeia abaixo)
```
Cadeia (faulthandler, C-traceback): `core/engine.py:90 → core/scanner.py:14 → core/detector.py:47 (import core.dl_backend) → numpy → SIGILL`. `/proc/cpuinfo`: flags `avx` **vazio**.

## Mitigações testadas e FALHARAM (todas rc 132)
- `OPENBLAS_CORETYPE=Nehalem` e `=Prescott`
- `NPY_DISABLE_CPU_FEATURES="AVX512F AVX512CD AVX2 AVX FMA3"`

→ Porque o crash é no **baseline compilado** do wheel musllinux do numpy (instruções always-on), **não** nos kernels despachados em runtime. Env vars só mexem no dispatch, não no baseline. Confirma: é o wheel PyPI assumindo um baseline de CPU que esta máquina não tem.

## 3 eixos — todos falham (este é o pior caso)
- **(a) graciosa?** NÃO — SIGILL duro, o engine inteiro morre, `BAREMETAL_SCAN: FAILED`, sem fallback.
- **(b) ajuda o user?** NÃO — rc 132, **mensagem ZERO**. O user vê só `import core.engine: FAILED`. **"G1-é-silent"** — exatamente o anti-padrão que a gente caça.
- **(c) resolve?** Não sozinho; e os toggles de runtime não resolvem (baseline).

## Impacto de tese (por que importa MUITO)
Bate direto no diferenciador **"leve pra hardware velho BR"**: o hardware MAIS velho (o alvo) é o mais provável de não ter o baseline do wheel → **crash silencioso sem explicação**. Risco de market-fit real, não cosmético.

## Fix recomendado (RO — Cursor desenha/implementa)
1. **Pre-flight de CPU ANTES de importar numpy/ML** (SIGILL não dá try/except — tem de ser pré-import): checar features mínimas vs as exigidas e, se faltar → **falhar gracioso com msg clara**: *"este CPU não tem &lt;feat&gt; exigido pelo wheel numpy; use o numpy da distro (`apk add py3-numpy` / `apt install python3-numpy`) ou build de fonte com `-Dcpu-baseline=min`."*
2. **ML opcional com fallback Python:** `core.dl_backend` já expõe `is_available`, mas `detector.py:47` importa eager → carrega numpy e mata **antes** do guard. Tornar o import de numpy/ML **lazy** (atrás de `is_available`/pre-flight) deixa o caminho não-ML (regex CPF/CNPJ em `core/brazilian_cpf`, `detector.py:46`) sobreviver = degradação graciosa.
3. **Remediação musl/alpine:** distro numpy (baseline conservador) em vez do wheel PyPI.

## Escopo / prioridade
alpine **não** está nos 4 hosts oficiais do gate #406 → **P2** (tese universal-platform / hardware-velho), **não bloqueia 1.7.4**. Mas é finding de alto valor pro diferenciador do produto. Nasceu do mini-bench smoke.

**Comment by FabioLeitao** [comment]
**✅ Resolvido no lab (29/jun) — receita + prova no metal.** _Comentário recomendatório (Claude = RO auditor; Cursor implementa)._

> _Correção: comentário anterior referenciava números que eram da minha lista interna de trabalho, não issues deste repo — removidos para não confundir._

## Root cause — corrige a hipótese do título
Não é o numpy-**core**. Com `-Dcpu-baseline=none` o core fica limpo (`objdump -d _multiarray_umath*.so | grep -c popcnt` → **0**). O assassino é o **`libscipy_openblas` (AVX/SSE4.2) que o PyPI EMBUTE na wheel** (pacote `scipy-openblas`, baseline com `popcnt`/AVX **incondicional**). Por isso `import scipy` e o matmul do numpy (BLAS) morrem; o `import numpy` puro pode passar e mascarar.

## Fix — provado no metal (alpine-emachines, host do gate de min-spec, sem AVX)
Build from-source contra o **OpenBLAS do SISTEMA** (Alpine `openblas`, `DYNAMIC_ARCH` = escolhe kernel no runtime = no-AVX-safe), **sem embutir** o scipy-openblas:
```sh
apk add build-base python3-dev py3-pip openblas openblas-dev gfortran binutils
pip install Cython meson-python ninja            # scipy tb: pybind11 pythran
pip uninstall -y scipy-openblas64 scipy-openblas32          # BLOQUEIO do AVX
pip wheel --no-binary numpy --no-build-isolation "numpy>=2.4.6" \
  -C setup-args=-Dblas=openblas -C setup-args=-Dlapack=openblas \
  -C setup-args=-Dcpu-baseline=none -C setup-args=-Dcpu-dispatch=none -w /out
# idem scipy/scikit-learn (-Dblas=openblas). Instalar a numpy from-source ANTES das build-deps de scipy.
```
Confirmar no log do build: `Run-time dependency scipy-openblas found: NO` + `openblas found: YES`.

## Prova
`numpy 2.5.0 + scipy 1.18.0 + scikit-learn 1.9.0 + pandas 3.0.4` — **RandomForest + TF-IDF rodando num Celeron 900 (2009, no-AVX, Alpine musl)**, wheels forjadas num i7-1365U (build-once-distribute). Receita completa + RCA documentados no lab.

## Armadilhas (custaram horas — documentar)
1. **`CFLAGS=-march=...` SABOTA** o baseline do numpy (deixa AVX no core) → **não usar**; deixar `-Dcpu-baseline=none` agir.
2. **Contaminação:** build-deps (`pythran`/`pybind11`) puxam numpy/scipy **binário do PyPI (`musllinux`, AVX)** → poison no `/out`; o pip no alvo **prefere o musllinux** sobre o `linux_x86_64` from-source limpo. Auditar `unzip -l <whl> | grep -c libscipy_openblas` (=0) + **remover as musllinux poison**.
3. Wheel fica **magra** (numpy 6MB vs 18MB) → alvo precisa `apk add openblas libgomp` (runtime dep) — ou `auditwheel repair` p/ self-contained.
4. **Falha SILENCIOSA (G1, já notado na issue):** instala/importa lindo na build-box AVX; só SIGILLa no alvo → **revalidar SEMPRE no metal**.

## AC sugerido
- [ ] `docs/plans/PLAN_NOAVX_WHEELHOUSE.md` com `<!-- plans-hub-summary: ... -->` + `python scripts/plans_hub_sync.py --write` + entry em `docs/plans/PLANS_TODO.md`
- [ ] Script oficial de build do **wheelhouse no-AVX** (container Alpine; receita acima) **+ gate de auditoria** que FALHA o build se `libscipy_openblas` ≠ 0 em qualquer wheel
- [ ] **Revalidação no metal** no `alpine-emachines` — install offline (`--no-index`) + smoke ML; objdump/import-test pegam a falha silent
- [ ] Doc dos runtime-deps (`openblas`/`libgomp`) no guia de install (transparência, anti-overclaim)
- [ ] **Installer detecta CPU** → capaz = wheel stock (AVX, +DL); no-AVX = wheelhouse (Regex+ML; DL degrada). Degrade **LOUD** + **FN-first** (nunca passa PII batido em silêncio) — doutrina perfis-por-capacidade
- Relacionado: #782 (irmão Rust/maturin, mesma filosofia de wheel-matrix)

**Comment by FabioLeitao** [comment]
**Reforço da matriz: musl + o eixo no-AVX (2026-07-09)**

A matriz Podman (glibc/musl/arm, host AVX) reforça este issue:
- **musl**: `scikit-learn` sem wheel musllinux → **compila do source** (Alpine + Void confirmados). No host AVX compila e passa (lento); no **Celeron no-AVX é onde o SIGILL deste issue morde**.
- Liga direto ao **wheelhouse no-AVX**: um `scikit-learn`/numpy/scipy **prebuilt musllinux + `cpu-baseline=none`** resolve os DOIS de uma vez — gap-de-wheel-musl **E** SIGILL no-AVX (sem compilar no alvo, sem instrução ilegal).
- ⚠️ Podman-no-host-AVX **não reproduz** o SIGILL (T14 tem AVX). O no-AVX segue exigindo **metal** (eMachines #821) ou VM com CPU-mask. A matriz cobre packaging (libc×arch); no-AVX = eixo capacidade, ortogonal.

Recomendo manter este issue como o hub do runner-floor no-AVX, com o wheelhouse (#82/#101) como o fix distribuível.

*(RO-auditor, recomendatório.)*

**Comment by FabioLeitao** [comment]
## Confirmado: ainda válida

Zero ocorrências de checagem SIGILL/AVX no repo — sem baseline de proteção contra esse crash silencioso.

*Verificado via A.I.I.D.C.O.B.P.P. v1.4 — Claude (Sonnet 5), auditor externo.*

**Comment by FabioLeitao** [comment]
**Enriquecimento pré-implementação (Claude, RO-auditor) — um gotcha real que pode desperdiçar uma
tentativa de fix.**

## O guard que já existe NÃO resolve isso

`core/dl_backend.py:20-33` já envolve `sentence_transformers`/`sklearn` em `try/except ImportError`.
Alguém pode achar que isso já cobre o caso e ir procurar o bug em outro lugar — **não cobre**.
`SIGILL` é um crash nativo/sinal do SO, não uma `ImportError` catável em Python. O processo morre
ANTES do `except` rodar. É por isso que o crash acontece mesmo com o guard existente no lugar.

## Cadeia exata (confirmada via `gh api`, não pelo clone local)

`core/detector.py` linha ~13: `from core.dl_backend import DLClassifier, is_available as dl_available`
— import de MÓDULO, incondicional, roda assim que `core.detector` é importado. `core.dl_backend`
por sua vez importa `sentence_transformers` (que puxa `torch`/`numpy`) dentro do próprio
`try/except`, mas o `except` não segura o SIGILL — o processo já morreu.

## Cuidado ao implementar o pre-flight

O pre-flight de CPU **não pode importar numpy pra testar numpy** (crasharia igual). Tem que checar
features via caminho que não toca numpy/torch — ex. ler `/proc/cpuinfo` direto (Linux) procurando
a flag `avx` na linha `flags`, ou `subprocess` pra `lscpu`. Só depois de confirmar que o CPU tem o
necessário é seguro deixar o import de `core.dl_backend`/`sentence_transformers` prosseguir.
